edition = "2023";

package cloud_databases_nl2code_eval_dbcodegen_proto;

import "eval_nl2code_request.proto";
import "eval_nl2code_response.proto";

option java_multiple_files = true;

service EvalCodeGenService {
  // Ping.
  rpc Ping(PingRequest) returns (EvalResponse) {
    // option deadline = 1800;
  }

  // Ping.
  rpc Connect(EvalConnectRequest) returns (EvalResponse) {
    // option deadline = 1800;
  }

  // Get_EvalInput.
  rpc EvalConfig(EvalConfigRequest) returns (EvalResponse) {
    // option deadline = 1800;
  }

  // ListEvalInputs.
  rpc ListEvalInputs(EvalInputRequest) returns (stream EvalInputRequest) {
    // option deadline = 1800;
  }

  // Evaluate.
  rpc Eval(stream EvalInputRequest) returns (EvalResponse) {
    // option deadline = 1800;
  }
}
